### PR TITLE
Document jupyterlab-gallery settings

### PR DESCRIPTION
**Pull request by krassowski**
## Reference Issues or PRs

Accompanies https://github.com/nebari-dev/nebari/pull/2501

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [x] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [x] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: docs/docs/explanations/advanced-custom-settings.md [hunk 1]
```diff
@@ -134,10 +134,15 @@ jupyterlab:
...
 :::note Note
-Currently only public git repositories are supported. Path location key should not start or end with trailing slash.
+Path location key should not start or end with trailing slash.
 You can configure JupyterLab to open in a location within the cloned repository by setting `preferred_dir` option within the `jupyterlab` group.
 :::
 
+See also `jupyterlab.gallery_settings` (documented below) which defers the cloning of repositories until user requests it and provides a rich presentation layer.
+
+:::warning
+While you could embed an access token in the URL to fetch from a private repository, please beware that it this token be accessed by the user - you should only use tightly scoped personal access tokens which you are comfortable to share with each of your users.
```
Comment (krassowski) [inline docs/docs/explanations/advanced-custom-settings.md:144]:
```suggestion
While you could embed an access token in the URL to fetch from a private repository, please beware that this token be accessed by each user - you should only use tightly scoped personal access tokens which you are comfortable to share with each of your users.
```
```suggestion
While you could embed an access token in the URL to fetch from a private repository, please beware that it this token be accessed by the user - you should only use tightly scoped personal access tokens which you are comfortable to share with each of your users.
```

File: docs/docs/explanations/advanced-custom-settings.md [hunk 2]
```diff
@@ -134,10 +134,15 @@ jupyterlab:
...
 :::note Note
-Currently only public git repositories are supported. Path location key should not start or end with trailing slash.
+Path location key should not start or end with trailing slash.
 You can configure JupyterLab to open in a location within the cloned repository by setting `preferred_dir` option within the `jupyterlab` group.
 :::
 
+See also `jupyterlab.gallery_settings` (documented below) which defers the cloning of repositories until user requests it and provides a rich presentation layer.
+
+:::warning
+While you could embed an access token in the URL to fetch from a private repository, please beware that this token be accessed by each user - you should only use tightly scoped personal access tokens which you are comfortable to share with each of your users.
```
Comment (Adam-D-Lewis) [inline docs/docs/explanations/advanced-custom-settings.md:144]: Typo - this token **will** be accessed by each user

**Comment by Adam-D-Lewis** [comment]
Small typo, but looks good otherwise.  Could you fix the typo and resolve the conflicts please?

File: docs/docs/explanations/advanced-custom-settings.md (hunk 2 above)
Comment (krassowski) [reply docs/docs/explanations/advanced-custom-settings.md:144]:
```suggestion
While you could embed an access token in the URL to fetch from a private repository, please beware that this token can be accessed by each user - you should only use tightly scoped personal access tokens which you are comfortable to share with each of your users.
```

**Review by Adam-D-Lewis — APPROVED** [review]
LGTM!

**Pull request merged**